interface_tester/collector.py

```diff
@@ -210,21 +217,20 @@ def _gather_test_cases_for_version(version_dir: Path, interface_name: str, versi
         # so we can import without tricks
         sys.path.append(str(interface_tests_dir))
 
-        for possible_test_file in interface_tests_dir.glob("*.py"):
-            # strip .py
-            module_name = str(possible_test_file.with_suffix("").name)
+        for role in Role:
+            module_name = "test_requirer" if role is Role.requirer else "test_provider"
             try:
-                importlib.import_module(module_name)
+                module = importlib.import_module(module_name)
             except ImportError as e:
-                logger.error(f"Failed to load module {possible_test_file}: {e}")
+                logger.error(f"Failed to load module {module_name}: {e}")
```

Comment:
It's generally recommended to use %s over f-string for logging (ref [here](https://docs.google.com/document/d/1H5W2oi7cuTaz86taTQiITv49jpJNKzzDZ2G25IQpE4w/edit#heading=h.r5obiqmhwoy3))

Reply:
This is not a charm, so I don't think it applies to this case.

Reply:
I agree that the rest of the charm style guide does not really apply, but the section about logging is relevant in any Python code